### ISSUE TITLE
Problems after installing browser themes.

### ISSUE THREAD

**Issue by EDSln** [Eclipse-Community/r3dfox-old#140]
There are two problems at once, with and without Aero. 
With the Windows 7 Base theme with any settings, the theme is not applied to the window title bar. 
r3dfox-128.0.2:
![2](https://github.com/user-attachments/assets/eacf3abf-6e98-4f21-b9f8-1469b3b062db)

It should be like Firefox 115 
![1](https://github.com/user-attachments/assets/b9b8e61c-31b4-45d8-9156-1e40e8c5921f)

With Aero the theme in the window title bar is applied after disabling the "Force enable caption bar transparency" setting, but the window control buttons, Minimize, Close are missing.
![3](https://github.com/user-attachments/assets/b59f9603-929f-4611-a3aa-b5702b698030)

If it's configurable in about:config, what are the settings?

**Comment by the-r3dacted** [comment]
The issue seems to be that -moz-window-titlebar coloring is not getting overridden by the lwtheme properly. I would need to figure out where this began.

**Comment by the-r3dacted** [comment]
I really need someone that can help me with my browser. I feel burnt out, and I don't want to do anything. I'm not sure how long I'll feel this way. It just hit me after I started to investigate the issue.

If you want to help fix it try previous releases to find the last to work, also disable r3dfox themes and styles as that seems to impact it, but I can probably easily fix it. Classic theme seems to be even more broken, so check it too. After that, if you know how to use browser toolbox, investigate why the default theme is overriding the lwtheme and see if it's potentially an easy CSS fix.

I'm going to lie down now, and not to sleep, or maybe I do end up napping idk.

**Comment by EDSln** [comment]
Disabling the r3dfox theme did not help.  I'll look for after which version the themes broke.  I'd like to help, but I'm completely clueless about CSS. :(

**Comment by jonm58** [comment]
on Windows 11(have Aero Glass)
![Windows_11](https://github.com/user-attachments/assets/863490ee-e48a-4b07-926c-0d6a51531a12)
btw,The button is just hidden.

**Comment by EDSln** [comment]
Checked many versions, problems vary from version to version, and depend on whether the window is expanded full screen or not, but everywhere after version 122 something doesn't work.
Beta 95% branded v122.0.1-2 - no problems.
Beta Aero v123.0.2 aero bug as on the screenshot below. basic in windowed mode bookmark bar without theme.
Beta Aero v123.0.2-2 aero bug as on the screenshot below. basic in window mode bookmark bar without theme.
124.0.2 aero bug as on the screenshot below. basic - in window mode bookmark bar without theme.
![123-124 bug aero](https://github.com/user-attachments/assets/93b523a7-30e1-4918-81ab-6d16d242079a)
124.0.3 aero - no problems. basic - full screen - no problems, in window mode - no theme. 
Release v125.0rc aero - no problems. basic - full screen - no problems, window mode - no theme. 
Release v125.0rc2 aero - no problems. basic - full screen no problems, window mode - no theme. 
Release v125rc3 aero - no buttons. basic - full screen - no problems, window mode - bookmarks panel without theme.
125.0.3 aero - no buttons. basic - full screen no problems, window mode - no theme. 
![124-125 basic no theme](https://github.com/user-attachments/assets/34254745-b738-45ed-b6ef-22a5224094d4)
126.0+ aero - no buttons. basic - window title bar without theme.

**Comment by jonm58** [comment]
![图片](https://github.com/user-attachments/assets/ab6c767c-9ffc-4597-bd0b-47417045ba5e)
I suspect that Firefox's browser theme is at the top level, causing problems with native patch controls.

**Comment by SoftwareType** [comment]
> on Windows 11(have Aero Glass) ![Windows_11](https://private-user-images.githubusercontent.com/73036321/351701112-863490ee-e48a-4b07-926c-0d6a51531a12.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4NDA3OTUsIm5iZiI6MTcyMTg0MDQ5NSwicGF0aCI6Ii83MzAzNjMyMS8zNTE3MDExMTItODYzNDkwZWUtZTQ4YS00YjA3LTkyNmMtMGQ2YTUxNTMxYTEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI0VDE3MDEzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5ZTY5OTA2NzE0MDE4MDBjOGEwYmZhNzU2MDRkOGJlMTFiYmIxNTAxMzk1NWM4ZDM0NzljYzkzOTgxZjQ2ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2B72_TmQUMVU8NkWHgsNsSJu7cdfm1jS9GEyTrE7cbQ) btw,The button is just hidden.

Without Aero Glass it also happens, but for some reason I suspect it is because of how Firefox handles the buttons. At default, Firefox always used their custom buttons, and, probably, those themes are trying to detect the buttons (and adjust the colors of it) but it fails

**Comment by KT907** [comment]
If you disable hardware acceleration and webrender in about:config, the buttons are displayed fine in third-party themes. Presumably, in this direction you need to look for compatibility errors.

**Comment by EDSln** [comment]
Yes, it looks like the problem is the wrong order of layers, there are buttons under the theme.
![not naming](https://github.com/user-attachments/assets/60da7604-5c9f-456e-88ee-e1695c4a9d71)

**Comment by the-r3dacted** [comment]
https://github.com/Eclipse-Community/r3dfox/issues/59

**Comment by EDSln** [comment]
Wouldn't it be easier to add an option, if possible, that completely reverts the Win 10 theme? Those who have the theme installed don't see the system window title bar anyway, and the fact that the window close buttons won't be from 7 is not so terrible.

**Comment by jonm58** [comment]
> Wouldn't it be easier to add an option, if possible, that completely reverts the Win 10 theme? Those who have the theme installed don't see the system window title bar anyway, 和 the fact that the window close buttons won't be from 7 is not so terrible.

 set widget.ev-native-controls-patch.override-win-version to 10,you can get Win 10 theme

**Comment by EDSln** [comment]
> set widget.ev-native-controls-patch.override-win-version to 10,you can get Win 10 theme

Thanks, no problem with Aero now. 
![1](https://github.com/user-attachments/assets/458622a7-e7b8-47b4-91a4-d84ad5c805b6)

But the Basic theme is still without the browser theme in the window title bar. Perhaps the problem is that Win10 simply can't have a Basic theme.

**Comment by the-r3dacted** [comment]
I can't fix this without many days of trying and probably failing to fix the cause. This is something I need external help with.

**Comment by J4CKED** [comment]
After the update (129.0.2 > 132.0), the min, max and close buttons disappeared.
As a temporary workaround, I installed this extension https://addons.mozilla.org/en-US/firefox/user/18029308/

**Comment by EDSln** [comment]
The buttons are behind the theme, so you can't see them, but if you click where they should be, the window will close or minimize.

**Comment by J4CKED** [comment]
Yes, I know, but I prefer to see them. I improved the buttons from the extension a little.
![Untitled-1](https://github.com/user-attachments/assets/dd805fd1-228e-4f80-b57a-88e85b58304a)

**Comment by EDSln** [comment]
In about:config find the widget.ev-native-controls-patch.override-win-version parameter and set it to 10, this will enable the buttons, but they will be styled 10 but in their place.

**Comment by J4CKED** [comment]
I did that, but those buttons are too big. I like the nice look.

**Comment by the-r3dacted** [comment]
I don't know how to fix this. I would need someone smart to figure this out and pr a fix.

**Comment by J4CKED** [comment]
I noticed that some windows 7 styles (msstyles) do not work well with the “force enable caption bar transparency” option. 
So I modified a favorite style to make it work with solid color.  I also made a second one, more gray-dark.
Firefox Arc Dark Theme and “Translucent design” option in 3rdfox. So far I like it.
![r3dfox](https://github.com/user-attachments/assets/765545fe-b69c-4d12-a802-1a35132c0b59)

**Comment by i486** [comment]
I just upgraded from 132 to 135.

Previously, I fixed the missing control buttons issue when using this [theme](https://addons.mozilla.org/en-US/firefox/addon/forest_/) by setting:

`widget.ev-native-controls-patch.override-win-version : 10`

But this setting no longer works here on 135

I noticed a "Modern UI" checkbox in Preferences, which restores the buttons, but it adds extra space above the tabs/buttons, making clicking difficult.

![Image](https://github.com/user-attachments/assets/acc75454-f6bc-46be-9696-1afc432ef138)

Any tips would be appreciated.

**Comment by the-r3dacted** [comment]
Ok, it was hell, but I got modern UI nearly perfect in 128.7. I was porting it back since 128 didn't have the easy option, and I discovered that it's completely broken when used with the light/dark themes. Except I didn't realize this was also an issue with later releases and spent over a day trying to match it to the initial 132 code without realizing it was broken to begin with.

I will now forward port the fixes to 135, and Probably make a new release somewhat soon of both 128.8 and 135.

**Comment by the-r3dacted** [comment]
btw this is the fix for the extra space issue in the code

![Image](https://github.com/user-attachments/assets/f80d4bda-ed7e-45ad-bd91-64ab4ddd00dc)

**Comment by the-r3dacted** [comment]
I might have a fix for this, it'll either be in next 135/136, or come when I get to 137/138

**Comment by the-r3dacted** [comment]
This should be fixed in 128.9.0, however I was unable to get the fix working in 135+ for now.
https://github.com/Eclipse-Community/r3dfox/releases/tag/v128.9.0

**Comment by the-r3dacted** [comment]
This should finally be fixed in the latest release.

https://github.com/Eclipse-Community/r3dfox/releases/tag/b136.0.4-1

**Comment by EDSln** [comment]
Will this be added in the next version of 128 ESR?

**Comment by the-r3dacted** [comment]
It should be fixed in ESR 128.9+

**Comment by i486** [comment]
> I just upgraded from 132 to 135.
> 
> Previously, I fixed the missing control buttons issue when using this [theme](https://addons.mozilla.org/en-US/firefox/addon/forest_/) by setting:
> 
> `widget.ev-native-controls-patch.override-win-version : 10`
> 
> But this setting no longer works here on 135
> 
> I noticed a "Modern UI" checkbox in Preferences, which restores the buttons, but it adds extra space above the tabs/buttons, making clicking difficult.
> 
> ![Image](https://github.com/user-attachments/assets/acc75454-f6bc-46be-9696-1afc432ef138)
> 
> Any tips would be appreciated.

I can confirm that this issue has been fixed since [Release v135.0.1-1-hotfix for Vista+](https://github.com/Eclipse-Community/r3dfox/releases/tag/v135.0.1-1-hotfix)
Thanks for the hard work.
